### DISCUSSION TITLE
Reactivity of state properties after an update

### DISCUSSION THREAD

**Discussion by franck102** [vuejs/pinia#1391] [Help and Questions] [answered]
Hi all, I am wondering if the following is a valid pattern, from the reactivity point of view, to load data from the server:
```javascript
export const myStore = defineStore('Booking', {
    state: () => ({
        items: {}
  )},
  actions: {
    loadData() {
        backend.get("/items").then(data => this.state.items = data)
    }
}
```

If a component used
```javascript
const { myItems } = storeToRefs(myStore);
```
.. before loadData was called, will "myItems"still be reactive after loadData replaces the state.items property?

Thanks!

**Comment by posva** [comment, marked as answer]
You mean `this.items` Instead of this.state.items ? If so, yes.

**Reply by franck102** [reply]
Thanks Posva,

This works as expected indeed, I had unrelated issues that made me unsure about the approach...
